src/api/tryGetFileTemporaryUrl.js

```diff
@@ -1,19 +1,49 @@
 /* @flow strict-local */
 import type { Auth } from './apiTypes';
+import { isUrlOnRealm } from '../utils/url';
 import getFileTemporaryUrl from './messages/getFileTemporaryUrl';
 
 /**
- * Get the complete authless URL to a realm uploaded file, returns null if that
- * fails.
+ * Like getFileTemporaryUrl, but on error returns null instead of throwing.
  *
- * See aslo jsdoc on getFileTemporaryUrl.
- *
- * @param href URL path to the resource.
- * @param auth Current user's auth information.
+ * Validates `url` to give getFileTemporaryUrl the kind of input it expects
+ * and returns null if validation fails.
  */
-export default async (href: string, auth: Auth): Promise<string | null> => {
+export default async (url: URL, auth: Auth): Promise<URL | null> => {
```

Comment:
LightboxScreen: Require URL object, not string, for props.route.params.src
    
    This means that we won't open the lightbox at all if the image's src
    isn't accepted by the URL parser. In that case, when the user taps
    an image in the message list, we give an error alert; see
    handleImage in src/webview/handleOutboundEvents.js.

```diff
-export default async (href: string, auth: Auth): Promise<URL | null> => {
-  // TODO: Have this function take this parsed form instead of raw string.
-  const parsedUrl = tryParseUrl(href, auth.realm);
+export default async (url: URL, auth: Auth): Promise<URL | null> => {
```

This seems like it'd be cleanest in a commit separate from the lightbox change.

I guess it has a couple of lightbox-related call sites that would grow a `new URL` call. But both of those already have a bunch of neighboring code that the main lightbox-related commit changes to be given a URL object instead of a string -- so if anything I think those diffs would also become simpler to read if they included changing those lines from `api.tryGetFileTemporaryUrl(new URL(url, auth.realm), …` to `api.tryGetFileTemporaryUrl(url, …` just like they do to other lines around them.